### PR TITLE
What if I will remove two lines of code?

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
We removed support for shared linking because of Rust. Actually, Rust is only an excuse for this removal, and we wanted to remove it nevertheless.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
We have to remove the shared ("splitted") build because of Rust.
Actually, I wanted to remove it a long time ago.

File: programs/CMakeLists.txt [hunk 1]
```diff
@@ -13,12 +13,6 @@ option (ENABLE_CLICKHOUSE_SERVER "Server mode (main mode)" ${ENABLE_CLICKHOUSE_A
 option (ENABLE_CLICKHOUSE_CLIENT "Client mode (interactive tui/shell that connects to the server)"
     ${ENABLE_CLICKHOUSE_ALL})
 
-# Don't create self-extracting clickhouse for split build
```
Comment (rschu1ze) [inline programs/CMakeLists.txt:16]: The option `SPLIT_SHARED_LIBRARIES` should be removed from the top-level CMakeLists and src/CMakeLists.txt.

**Comment by rschu1ze** [comment]
I like that this reduces build and CI complexity quite a bit - a good side effect of Rust!

I never used shared builds for my dev so I don't care. It would be interesting to know if others do. Perhaps make a poll in Slack/TG? If it shared builds are popular, they could be kept alive at least in CMake (but still removed from CI)...

**Comment by qoega** [comment]
Checks are too green. @rschu1ze are you ready to merge?

**Comment by rschu1ze** [comment]
Let me have another look if something is missing here. Afterwards we can merge.

**Review by rschu1ze — APPROVED** [review]

**Comment by azat** [comment]
Removing build with shared libs (and this what this PR does), is a sad news.

>We have to remove the shared ("splitted") build because of Rust.

And what was the problem with Rust and shared build?

**Comment by alexey-milovidov** [comment]
@azat No problem at all, Rust is just an excuse for removing shared linking.

**Comment by alexey-milovidov** [comment]
So, if you don't like it, you will blame Rust :laughing:

**Comment by alexey-milovidov** [comment]
@azat Maybe you can reintroduce it, but with as minimal changes in CMake as possible.
So, there will be no notion of "splitted" build.

**Comment by alexey-milovidov** [comment]
@azat Most likely, we will have to re-introduce shared linking because of Python (to build a Python module), so it still has a chance.

**Pull request merged**